### DISCUSSION TITLE
Scientific notation for large numbers

### DISCUSSION THREAD

**Discussion by rms95** [nlohmann/json#3958] [Ideas]
We have a JSON file with large numbers (doubles in C++), like 10e+14.
With the current implementation this is written as 100000000000000, which is not very readable.

Would it be possible to make the following two limits configurable?
```C++
    // Format the buffer like printf("%.*g", prec, value)
    constexpr int kMinExp = -4;
    // Use digits10 here to increase compatibility with version 2.
    constexpr int kMaxExp = std::numeric_limits<FloatType>::digits10;
```

**Comment by nlohmann** [comment]
No, this is currently not configurable, and there is currently no concept to achieve this.
